src/client/LiveStream.js

```diff
@@ -807,51 +805,49 @@ exports.StreamStopSession = async function({name}) {
         error: "no active streams - must create a stream first"
       };
     }
-    let edgeMeta = await this.ContentObjectMetadata({
-      libraryId,
-      objectId,
-      writeToken: edgeWriteToken
-    });
 
-    // Stop the LRO if running
-    let status = await this.StreamStatus({name});
-    if(status.state != "terminated") {
-      console.log("STOPPING");
-      try {
-        await this.CallBitcodeMethod({
-          libraryId: status.library_id,
-          objectId: status.object_id,
-          writeToken: status.edge_write_token,
-          method: "/live/stop/" + status.tlro,
-          constant: false
-        });
-      } catch(error) {
-        // The /call/lro/stop API returns empty response
-        // console.log("LRO Stop (failed): ", error);
-      }
+    let edgeMeta, status;
+    try {
+      edgeMeta = await this.ContentObjectMetadata({
+        libraryId,
+        objectId,
+        writeToken: edgeWriteToken
+      });
 
-      // Wait until LRO is terminated
-      let tries = 10;
-      while(status.state != "stopped" && tries-- > 0) {
-        console.log("Wait to terminate - ", status.state);
-        await Sleep(1000);
-        status = await this.StreamStatus({name});
+      status = await this.StreamStatus({name});
+
+      if(status.state !== "stopped") {
+        return {
+          state: status.state,
+          error: "Stream must be stopped before terminating"
+        }
       }
-      console.log("Status after stop - ", status.state);
+    } catch(error) {
+      console.warn(`Unable to retrieve metadata for edge write token ${edgeWriteToken}`);
 
-      if(tries <= 0) {
-        console.log("Failed to stop");
-        return status;
+      // If token doesn't return metadata, generate new token
+      if(!edgeMeta) {
+        let response = await this.EditContentObject({
+          libraryId: libraryId,
```

Comment:
We should create the new write token in all cases.
This is because the edge write token has a ton of metadata in it - as created by the live recorder (the list of all video and audio parts and metadata for each part, ...).

I think we should do nothing when edge_meta doesn't work.
And if it does work then we can check the status, like you already to, and then discard/delete the edge write token (though I am not sure if we have a client method to discard a write token)

Then just open the new write token in all cases and write the status fields, then finalize and publish.

Reply:
It doesn't look like there's a client method to discard a write token. Is there a contract method for this?